File: cpp/gabaritos/gabarito6.cpp
#include <iostream>
#include <map>
#include <string>
#include <sstream>

int main() {
    std::string frase = "o rato roeu a roupa do rei de roma";
    std::stringstream ss(frase);
    std::string palavra;
    std::map<std::string, int> contagem;
    while (ss >> palavra) {
        contagem[palavra]++;
    }
    for (const auto& par : contagem) {
        std::cout << par.first << ": " << par.second << std::endl;
    }
    return 0;
}

File: csharp/gabaritos/gabarito6.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        string frase = "o rato roeu a roupa do rei de roma";
        string[] palavras = frase.Split(' ');
        Dictionary<string, int> contagem = new Dictionary<string, int>();
        foreach (string palavra in palavras)
        {
            if (contagem.ContainsKey(palavra))
            {
                contagem[palavra]++;
            }
            else
            {
                contagem[palavra] = 1;
            }
        }
        foreach (var kvp in contagem)
        {
            Console.WriteLine($"{kvp.Key}: {kvp.Value}");
        }
    }
}